### PR TITLE
fix: address clawpatch findings

### PR DESCRIPTION
**Pull request by chapati23**
## Summary
- fix six clawpatch findings across indexer, dashboard health/market-hours handling, and deterministic test fixtures
- make Wormhole delivered rollups rely on merged transfer data instead of destination manager metadata when source data exists
- record the new clawpatch findings as fixed in tracked .clawpatch state

## Test plan
- pnpm agent:quality-gate --run
- pre-push agent quality gate via git push
- Chrome DevTools browser verification on http://localhost:3000 with no console errors

---

> [!NOTE]
> **Medium Risk**
> Touches Wormhole bridge handler rollup/pairing logic and dashboard health/market-hours status calculation, which can affect user-visible correctness and indexed aggregates; mitigated by added regression tests. Remaining changes are mostly test infrastructure and build-script hygiene.
> 
> **Overview**
> Fixes Wormhole bridge indexing edge cases: pending scratch rows are now **always deleted** (no optional chaining), and `TransferRedeemed` delivery rollups no longer depend on destination manifest lookup when merged transfer data is sufficient (while still skipping `UNKNOWN` tokens). Adds regression coverage for scratch draining and for delivery counting when the destination manager is missing.
> 
> Corrects dashboard status logic by deriving `MarketHoursPill` open/closed from the on-chain MARKET_HOURS breaker state (with weekday-closure tests), and by treating `oracleOk === false` as `CRITICAL` even when timestamps are fresh (with new health tests). Makes shared network test fixtures deterministic via `TEST_NOW` (plus fixture tests) and updates volume chart tests that require real-time windows.
> 
> Improves indexer test infrastructure by making the HTTP test RPC server more robust to port conflicts (fallback to ephemeral port) and updates `indexer-envio`’s `clean` script to `tsc --build --clean`. Records new `.clawpatch` finding/feature statuses and adds the corresponding finding JSON artifacts.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 9c5c8e5cea7377bfda363e5adfe17bf751466fe0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by chapati23** [comment]
@claude review

**Comment by chapati23** [comment]
Fixed in db62995e — added a health regression test confirming a fresh oracle with oracleOk=false stays CRITICAL even when the weekend branch is mocked active.

File: ui-dashboard/src/components/market-hours-pill.tsx [hunk 1]
```diff
@@ -127,17 +135,22 @@ export function MarketHoursPill({ pool }: Props): React.ReactElement | null {
...
       <span
         className="inline-flex items-center gap-1 rounded bg-slate-800/80 px-1.5 py-0.5 text-xs cursor-help"
         title={tooltip}
       >
         <span className="text-slate-300 font-medium">Market Closed</span>
-        <span className="text-slate-500">·</span>
-        <span className="font-mono text-slate-300">
-          {formatHoursMinutes(secondsUntilTransition)} until open
-        </span>
+        {!breakerClosed && (
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/components/market-hours-pill.tsx]: Fixed in c8b1e8c3 — `MarketHoursPill` now distinguishes calendar closure from breaker closure and keeps the “until open” countdown when the calendar is closed, even if MARKET_HOURS is also tripped. Added a Saturday breaker-trip regression test.

File: ui-dashboard/src/components/market-hours-pill.tsx [hunk 2]
```diff
@@ -127,17 +135,22 @@ export function MarketHoursPill({ pool }: Props): React.ReactElement | null {
...
+          <>
+            <span className="text-slate-500">·</span>
+            <span className="font-mono text-slate-300">
+              {formatHoursMinutes(secondsUntilTransition)} until open
+            </span>
+          </>
+        )}
```
[the remark this answers is not in the document]
Comment (chapati23) [reply ui-dashboard/src/components/market-hours-pill.tsx:154]: Fixed in c8b1e8c3 — same root cause as the Codex thread: the countdown now keys off calendar closure, so FX weekends keep the reopen timer even when MARKET_HOURS is tripped. Added a Saturday tripped-breaker regression test.
